### ISSUE TITLE
I try to install rmagick gem  but  i not able install  build issue ...

### ISSUE THREAD

**Issue by rsurendraidexcel** [rmagick/rmagick#1228]
Gemfile

gem "sinatra"
gem 'rmagick'
gem 'gruff', '~> 0.10.0'

$ bundle install 
i got the following issue   how to fixed please help me 

**issue 1**
![rmagic_install_issue2](https://user-images.githubusercontent.com/7713535/89108702-1bfd0a00-d458-11ea-811a-0749f58bd77c.png)

**issue 2**
![rmagic_2_isuue](https://user-images.githubusercontent.com/7713535/89108710-31723400-d458-11ea-821b-3b52783a431a.png)

**Comment by Watson1978** [comment]
Your build logs indicate that `SetImageGray()` and `SetMagickAlignedMemoryMethods()` are missing. This is odd.
Because these APIs are existing in ImageMagick 7.0.9.

```
checking for SetImageGray() in assert.h,ctype.h,stdio.h,stdlib.h,math.h,time.h,sys/types.h,MagickCore/MagickCore.h... yes
checking for SetMagickAlignedMemoryMethods() in assert.h,ctype.h,stdio.h,stdlib.h,math.h,time.h,sys/types.h,MagickCore/MagickCore.h... yes
```

@rsurendraidexcel In order to solve your problem, I think we need to know how you installed ImageMagick 7.0.9 at least.

**Comment by mockdeep** [comment]
@rsurendraidexcel were you able to solve your issue? If not, could you please provide more information about how you installed ImageMagick?

**Comment by dlemstra** [comment]
This looks like a similar error reported in #1237. Would be nice if we could reproduce this inside a docker container.

**Comment by Watson1978** [comment]
I made a reproducible Dockerfile in https://gist.github.com/Watson1978/dbeb913aedd3c8e7b87bd9c8c7524b09

You might reproduce this issue via following commands.

```
$ mkdir ~/work
$ cd ~/work
$ wget https://gist.githubusercontent.com/Watson1978/dbeb913aedd3c8e7b87bd9c8c7524b09/raw/2271fbfa52631258b0ea5df98c727d96a7ab15bf/Dockerfile

$ docker build -t rmagick-ubuntu-20_04 .
$ git clone git@github.com:rmagick/rmagick.git

$ docker run -v ~/work/rmagick:/opt/rmagick --rm -it rmagick-ubuntu-20_04 bash

# gem install bundler
# bundle
# rake
...
ompiling ../../../../ext/RMagick/rmimage.c
../../../../ext/RMagick/rmimage.c: In function 'Image_gray_q':
../../../../ext/RMagick/rmimage.c:7552:32: error: 'IsGrayImage' undeclared (first use in this function); did you mean 'DrawImage'?
 7552 |     return has_attribute(self, IsGrayImage);
      |                                ^~~~~~~~~~~
      |                                DrawImage
../../../../ext/RMagick/rmimage.c:7552:32: note: each undeclared identifier is reported only once for each function it appears in
make: *** [Makefile:245: rmimage.o] Error 1
rake aborted!
Command failed with status (2): [/usr/bin/make...]
/var/lib/gems/2.7.0/gems/rake-compiler-1.1.1/lib/rake/extensiontask.rb:167:in `block (2 levels) in define_compile_tasks'
/var/lib/gems/2.7.0/gems/rake-compiler-1.1.1/lib/rake/extensiontask.rb:166:in `block in define_compile_tasks'
/usr/share/rubygems-integration/all/gems/rake-13.0.1/exe/rake:27:in `<top (required)>'
Tasks: TOP => default => spec => compile => compile:x86_64-linux-gnu => compile:RMagick2:x86_64-linux-gnu => copy:RMagick2:x86_64-linux-gnu:2.7.0 => tmp/x86_64-linux-gnu/RMagick2/2.7.0/RMagick2.so
(See full trace by running task with --trace)
```

**Comment by dlemstra** [comment]
Took me a while to find it but I tracked down the issue. I hacked one of the ruby files to display the error message and I got a linker error. Then I found this issue https://github.com/libvips/libvips/issues/418#issuecomment-207514248. The problem is that in the container ImageMagick is build with `--with-wmf`. This requires to also link the `libMagickWand` and then `have_func` fails because this is not linked. One of the solutions reported in the libvips issue was to use `--with-modules`. In the next version of ImageMagick we will only enable `--with-wmf` by default when `--with-modules` is enabled.
